### ISSUE TITLE
Weather file is not read when the period is longer than one year

### ISSUE THREAD

**Issue by wwolff7** [daquinterop/Py_DSSATTools#2]
Hello, I have observed that the simulations are not performaded rightly when there are more than one .WTH file. For example, I runned a simulations from 2016-10-11 to 2017-04-11, checking the results, I observed that the outputs are related to the period from 2016-10-11 to 2016-12-31. I pressume would be necessary to join the files *1610.WTH and *1704.WTH. Checking WARNING.OUT file it is possible to see:

```
 IPWTH   YEAR DOY = 2017   1
 Weather record not found for YR DOY: 2017   1
   /tmp/dssatruns/Weather/DPOE1608.WTH
 Simulation will end.
```

**Comment by daquinterop** [comment]
Hi Wagner, I think I came into the same issue a few weeks ago, I already solved it (I think) but I forgot to push the changes. I just push the [last commit](https://github.com/daquinterop/Py_DSSATTools/commit/34522365ae3c7476ab4e9670c51c7f84e2e0db21), so would you please reinstall the library from the last version in the repository?? Please let me know if that fixed the issue to upload the last version to PyPI. Thanks again for reporting this issue.

**Comment by wwolff7** [comment]
Hello Diego,

Now is working well, feel free to publish the new release.

How about DSSAT calibration? Do you have any idea how to do that?

Besta regards!

Wagner


Em sex, 6 de jan de 2023 15:43, Diego Quintero ***@***.***>
escreveu:

**Comment by daquinterop** [comment]
Sure!! I started the development of this library so that it can be used for Bayesian calibration. Right now I'm kinda working on that. Feel free to message me if you wanna discuss about calibration.

**Comment by liao007** [comment]
Hello Diego,

Your python project on DSSAT is great. I am going to use these tools to conduct DSSAT modeling. Yet calibration is neccesary. I would like to know how is the progress on the calibration library? Are you still working on the development of the calibration library? If yes, When the library will be published? 

Thank you very much!

Best regards!
Liao

**Comment by daquinterop** [comment]
> Hello Diego,
> 
> Your python project on DSSAT is great. I am going to use these tools to conduct DSSAT modeling. Yet calibration is neccesary. I would like to know how is the progress on the calibration library? Are you still working on the development of the calibration library? If yes, When the library will be published?
> 
> Thank you very much!
> 
> Best regards! Liao

Hello! I stopped working on any calibration tools. I reached the conclusion that it is not worth it. Python already has several calibration utilities (scipy.optimize, pymc, ax, etc.) that can be easily integrated in any calibration problem, including calibrating the DSSAT model using this library. Besides, any "easy to use" calibration tool could make the user ignore important factors in their specific calibration problem. It is better to take the time to understand the problem to solve, and then decide the best tools and methods to perform your calibration. If your calibrating cultivar parameters  check [this paper](https://www.sciencedirect.com/science/article/abs/pii/S1364815221002486), and try to use basic optimization tools such as scipy.optimize. Good luck with your project!
